internal/goldendataset/span_generator.go

```diff
@@ -0,0 +1,497 @@
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package goldendataset
+
+import (
+	"fmt"
+	"io"
+	"time"
+
+	otlpcommon "github.com/open-telemetry/opentelemetry-proto/gen/go/common/v1"
+	otlptrace "github.com/open-telemetry/opentelemetry-proto/gen/go/trace/v1"
+
+	"go.opentelemetry.io/collector/translator/conventions"
+)
+
+var statusCodeMap = map[PICTInputStatus]otlptrace.Status_StatusCode{
+	SpanStatusOk:                 otlptrace.Status_Ok,
+	SpanStatusCancelled:          otlptrace.Status_Cancelled,
+	SpanStatusUnknownError:       otlptrace.Status_UnknownError,
+	SpanStatusInvalidArgument:    otlptrace.Status_InvalidArgument,
+	SpanStatusDeadlineExceeded:   otlptrace.Status_DeadlineExceeded,
+	SpanStatusNotFound:           otlptrace.Status_NotFound,
+	SpanStatusAlreadyExists:      otlptrace.Status_AlreadyExists,
+	SpanStatusPermissionDenied:   otlptrace.Status_PermissionDenied,
+	SpanStatusResourceExhausted:  otlptrace.Status_ResourceExhausted,
+	SpanStatusFailedPrecondition: otlptrace.Status_FailedPrecondition,
+	SpanStatusAborted:            otlptrace.Status_Aborted,
+	SpanStatusOutOfRange:         otlptrace.Status_OutOfRange,
+	SpanStatusUnimplemented:      otlptrace.Status_Unimplemented,
+	SpanStatusInternalError:      otlptrace.Status_InternalError,
+	SpanStatusUnavailable:        otlptrace.Status_Unavailable,
+	SpanStatusDataLoss:           otlptrace.Status_DataLoss,
+	SpanStatusUnauthenticated:    otlptrace.Status_Unauthenticated,
+}
+
+var statusMsgMap = map[PICTInputStatus]string{
+	SpanStatusOk:                 "",
+	SpanStatusCancelled:          "Cancellation received",
+	SpanStatusUnknownError:       "",
+	SpanStatusInvalidArgument:    "parameter is required",
+	SpanStatusDeadlineExceeded:   "timed out after 30002 ms",
+	SpanStatusNotFound:           "/dragons/RomanianLonghorn not found",
+	SpanStatusAlreadyExists:      "/dragons/Drogon already exists",
+	SpanStatusPermissionDenied:   "tlannister does not have write permission",
+	SpanStatusResourceExhausted:  "ResourceExhausted",
+	SpanStatusFailedPrecondition: "33a64df551425fcc55e4d42a148795d9f25f89d4 has been edited",
+	SpanStatusAborted:            "",
+	SpanStatusOutOfRange:         "Range Not Satisfiable",
+	SpanStatusUnimplemented:      "Unimplemented",
+	SpanStatusInternalError:      "java.lang.NullPointerException",
+	SpanStatusUnavailable:        "RecommendationService is currently unavailable",
+	SpanStatusDataLoss:           "",
+	SpanStatusUnauthenticated:    "nstark is unknown user",
+}
+
+func GenerateSpans(count int, startPos int, pictFile string, random io.Reader) ([]*otlptrace.Span, int, error) {
```

Comment:
Can we document int parameters and return types? It is non-obvious unless you read the function body.

Reply:
Documented all public functions